api/router.mjs

```diff
@@ -0,0 +1,42 @@
+import { NOT_FOUND } from './defs.mjs'
+import * as db from './db.mjs'
+
+const handlers = {}
+const setter = (p) => (_, k, v) => (handlers[`${p}/${k}`] = v)
+const getter = (p) => (cache, k) => cache[k] || (cache[k] = ROUTE(`${p}/${k}`))
+const ROUTE = (p) => new Proxy({}, { get: getter(p), set: setter(p) })
+
+export const GET = ROUTE('GET/api')
+export const POST = ROUTE('POST/api')
+
+export const withUser = (fn) => async (params) => {
+  if (!params.session) return new Response('No Session', UNAUTHORIZED)
+  params.user = await db.get(params.session)
+  if (!params.user) return new Response('Bad Session', UNAUTHORIZED)
+  // ...maybe remove session cookie in that case
+  return fn(params)
+}
+
+export const withRoom = (fn) => async (params) => {
+  const { ROOM } = params.env
+  ROOM.idFromName('nan')
+  params.room = await ROOM.get(id)
+  return fn(params)
+}
```

Comment:
not yet tested, but this will add the websocket `durable object` once I implemented it